public/javascripts/main.js

```diff
@@ -1,19 +1,51 @@
 $(function(){
     $('#select-all').click(function(event) {
-        if(this.checked) {
-            // Iterate each checkbox
-            $('.schedule:checkbox').each(function() {
-                this.checked = true;
-            });
-        }
-        else {
-            $(':checkbox').each(function() {
+        if (this.checked) {
+            smartSelectAll("Rotterdam");
+        } else {
+            $(':checkbox').each(function () {
                 this.checked = false;
             });
         }
     });
 });
 
+/*
+Will perform "smart" select all. Will only check one item per day.
+1. If there's already a check for that day, it will remain unchanged.
+2. If there's only one meal for that day, it will be checked regardless of location.
+3. It will always go for the default location, which is Rotterdam. Unless we implement a "preferred location" per user.
+4. If there's more than one meal with the same location, it will check the first meal in the list.
+*/
+function smartSelectAll(defaultLocation) {
+    var dates = $("input[data-date]").map(function() {
+        return $(this).attr("data-date");
+    }).get();
+    var uniqueDates = jQuery.unique(dates);
+
+    uniqueDates.forEach(function(date) {
+        var hasCheck = false;
+        var inputsByDate = $("input[data-date='" + date + "']");
+
+        var checkedInputsByDate = $("input[data-date='" + date + "']:checked");
+        if(checkedInputsByDate.length == 0) {
+            inputsByDate.each(function() {
+                if(inputsByDate.length == 1) {
+                    this.checked = true;
+                } else {
+                    var location = this.parentNode.parentNode.children[5].textContent;
```

Comment:
can you query the desired parent using `$(this).parents(...)`? Would be more descriptive and less prone to breaking.